components/features/Signup/AssesseeSignup/AssesseeDetails/AssesseeDetails.tsx

```diff
@@ -0,0 +1,120 @@
+import { InputField } from "@components/shared/forms/InputField";
+import { OdiLogo } from "@components/shared/svg/OdiLogo";
+import { useAssesseeSignupStoreContext } from "@context/Signup/AssesseeSignupStoreContext";
+import styles from "./AssesseeDetails.module.css";
+import Flatpickr from "react-flatpickr";
+
+import React, { useEffect, useMemo, useRef } from "react";
+import { Button } from "@components/shared/elements/Button";
+import { SigninNotice } from "@components/shared/forms/SigninNotice";
+import { DateField } from "@components/shared/forms/DateField";
+import { emptyValidator } from "@utils/validators/emptyValidator";
+import { dateValidator } from "@utils/validators/dateValidator/dateValidator";
+import { PhoneField } from "@components/shared/forms/PhoneField";
+import { useAssesseeSignupStepContext } from "@context/Signup/AssesseeSignupStepContext";
+
+const AssesseeDetails = () => {
+  const { selectStep, selectedId } = useAssesseeSignupStepContext();
+  const { data, errors, setValue, setError } = useAssesseeSignupStoreContext();
+  const firstNameRef = useRef<HTMLInputElement>(null);
+  const lastNameRef = useRef<HTMLInputElement>(null);
+  const dateOfBirthRef = useRef<Flatpickr>(null);
+
+  const inputRefs = useMemo(
```

Comment:
Hello Alex, can you walk me through your thought process on using useMemo instead of useRef here :)

Reply:
Hello Rashad, I intend to use the useMemo hook so that the state is saved after rerenders.  I decided not to use useRef as to not add more boilerplate, removing .current I think can simplify our codebase better. 😀